Cargo.toml

```diff
@@ -7,6 +7,7 @@ members = [
     # FIXME: getting "multiple matching crates for `diesel`" because of the shared build directory
     # "diesel_compile_tests",
     "diesel_tests",
+    "diesel_infer_schema",
```

Comment:
Moving this into a separate, public crate is interesting. I'm not sure I agree, but I don't see many reasons against it.